[BUG] OpenAiTokenizer throws NullPointerException on tool arguments that are null

<!-- Please provide as many details as possible, this will help us to deliver a fix as soon as possible. Thank you! -->

**Describe the bug**
<!-- A clear and concise description of what the bug is. -->
When estimating tokens with the OpenAiTokenizer, a NullPointerException is thrown if the estimated message contains a tool execution request with a tool argument that has a null value.

**Log and Stack trace**
<!-- Please provide a log and a stack trace (with exception), if applicable. -->
```
Caused by: java.lang.NullPointerException: Cannot invoke "Object.toString()" because the return value of "java.util.Map$Entry.getValue()" is null
  at dev.langchain4j.model.openai.OpenAiTokenizer.estimateTokenCountIn(OpenAiTokenizer.java:164)
  at dev.langchain4j.model.openai.OpenAiTokenizer.estimateTokenCountInMessage(OpenAiTokenizer.java:96)
  at dev.langchain4j.model.openai.OpenAiTokenizer.estimateTokenCountInMessages(OpenAiTokenizer.java:199)
  at dev.langchain4j.memory.chat.TokenWindowChatMemory.ensureCapacity(TokenWindowChatMemory.java:99)
  at dev.langchain4j.memory.chat.TokenWindowChatMemory.add(TokenWindowChatMemory.java:75)
  at dev.langchain4j.service.tool.ToolService.executeInferenceAndToolsLoop(ToolService.java:126)
  at dev.langchain4j.service.DefaultAiServices$1.invoke(DefaultAiServices.java:228)
  at jdk.proxy2/jdk.proxy2.$Proxy121.chat(Unknown Source)
  at 
...
```

**To Reproduce**
<!-- Please provide a relevant code snippets to reproduce this bug. -->
Define an AiService with multiple tools, one of which has a parameter that the LLM can set to `null`. The AiService needs a `TokenWindowChatMemory` with the `OpenAiTokenizer`.

**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->
Not see an error but instead get a token estimation (and consequently get an answer from the AiService that uses the Tokenizer).

**Please complete the following information:**
- LangChain4j version: 1.0.0-beta3 (I checked, the new OpenAiTokenCountEstimator in version 1.0.0-rc still contains the problematic code snippet)
- Java version: 17

--- a/langchain4j-azure-open-ai/src/main/java/dev/langchain4j/model/azure/AzureOpenAiTokenCountEstimator.java
+++ b/langchain4j-azure-open-ai/src/main/java/dev/langchain4j/model/azure/AzureOpenAiTokenCountEstimator.java
@@ -20,6 +20,7 @@ import java.util.Optional;
 import java.util.function.Supplier;
 
 import static dev.langchain4j.internal.Exceptions.illegalArgument;
+import static dev.langchain4j.internal.Utils.isNullOrBlank;
 import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
 import static dev.langchain4j.model.azure.AzureOpenAiChatModelName.GPT_3_5_TURBO;
 import static dev.langchain4j.model.azure.AzureOpenAiChatModelName.GPT_3_5_TURBO_0301;
@@ -157,16 +158,19 @@ public class AzureOpenAiTokenCountEstimator implements TokenCountEstimator {
                     tokenCount += 7;
                     tokenCount += estimateTokenCountInText(toolExecutionRequest.name());
 
-                    Map<?, ?> arguments;
+                    if (isNullOrBlank(toolExecutionRequest.arguments())) {
+                        continue;
+                    }
+
                     try {
-                        arguments = OBJECT_MAPPER.readValue(toolExecutionRequest.arguments(), Map.class);
+                        Map<?, ?> arguments = OBJECT_MAPPER.readValue(toolExecutionRequest.arguments(), Map.class);
+                        for (Map.Entry<?, ?> argument : arguments.entrySet()) {
+                            tokenCount += 2;
+                            tokenCount += estimateTokenCountInText(String.valueOf(argument.getKey()));
+                            tokenCount += estimateTokenCountInText(String.valueOf(argument.getValue()));
+                        }
                     } catch (JsonProcessingException e) {
                         throw new RuntimeException(e);
-                    }
-                    for (Map.Entry<?, ?> argument : arguments.entrySet()) {
-                        tokenCount += 2;
-                        tokenCount += estimateTokenCountInText(argument.getKey().toString());
-                        tokenCount += estimateTokenCountInText(argument.getValue().toString());
                     }
                 }
             }

--- a/langchain4j-open-ai/src/main/java/dev/langchain4j/model/openai/OpenAiTokenCountEstimator.java
+++ b/langchain4j-open-ai/src/main/java/dev/langchain4j/model/openai/OpenAiTokenCountEstimator.java
@@ -22,6 +22,7 @@ import java.util.function.Supplier;
 
 import static com.knuddels.jtokkit.api.EncodingType.O200K_BASE;
 import static dev.langchain4j.internal.Exceptions.illegalArgument;
+import static dev.langchain4j.internal.Utils.isNullOrBlank;
 import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
 
 /**
@@ -138,16 +139,19 @@ public class OpenAiTokenCountEstimator implements TokenCountEstimator {
                     tokenCount += 7;
                     tokenCount += estimateTokenCountInText(toolExecutionRequest.name());
 
-                    Map<?, ?> arguments;
+                    if (isNullOrBlank(toolExecutionRequest.arguments())) {
+                        continue;
+                    }
+
                     try {
-                        arguments = OBJECT_MAPPER.readValue(toolExecutionRequest.arguments(), Map.class);
+                        Map<?, ?> arguments = OBJECT_MAPPER.readValue(toolExecutionRequest.arguments(), Map.class);
+                        for (Map.Entry<?, ?> argument : arguments.entrySet()) {
+                            tokenCount += 2;
+                            tokenCount += estimateTokenCountInText(String.valueOf(argument.getKey()));
+                            tokenCount += estimateTokenCountInText(String.valueOf(argument.getValue()));
+                        }
                     } catch (JsonProcessingException e) {
                         throw new RuntimeException(e);
-                    }
-                    for (Map.Entry<?, ?> argument : arguments.entrySet()) {
-                        tokenCount += 2;
-                        tokenCount += estimateTokenCountInText(argument.getKey().toString());
-                        tokenCount += estimateTokenCountInText(argument.getValue().toString());
                     }
                 }
             }

--- a/langchain4j-open-ai/src/test/java/dev/langchain4j/model/openai/OpenAiTokenCountEstimatorTest.java
+++ b/langchain4j-open-ai/src/test/java/dev/langchain4j/model/openai/OpenAiTokenCountEstimatorTest.java
@@ -1,9 +1,13 @@
 package dev.langchain4j.model.openai;
 
+import dev.langchain4j.agent.tool.ToolExecutionRequest;
+import dev.langchain4j.data.message.AiMessage;
 import dev.langchain4j.model.TokenCountEstimator;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.EnumSource;
+import org.junit.jupiter.params.provider.NullSource;
+import org.junit.jupiter.params.provider.ValueSource;
 
 import java.util.ArrayList;
 import java.util.List;
@@ -46,6 +50,51 @@ class OpenAiTokenCountEstimatorTest {
         assertThat(tokenCountEstimator.estimateTokenCountInText("\n")).isEqualTo(1);
         assertThat(tokenCountEstimator.estimateTokenCountInText("\n\n")).isEqualTo(1);
         assertThat(tokenCountEstimator.estimateTokenCountInText("\n \n\n")).isEqualTo(2);
+    }
+
+    @Test
+    void should_count_tokens_in_ai_message_with_null_tool_argument() {
+
+        // given
+        ToolExecutionRequest toolExecutionRequest = ToolExecutionRequest.builder()
+                .id("tool-id")
+                .name("tool-name")
+                .arguments("""
+                        {
+                            "name": "Klaus",
+                            "address": null
+                        }
+                        """)
+                .build();
+
+        AiMessage aiMessage = AiMessage.from(toolExecutionRequest, toolExecutionRequest); // duplicate on purpose
+
+        // when
+        int tokenCount = tokenCountEstimator.estimateTokenCountInMessage(aiMessage);
+
+        // then
+        assertThat(tokenCount).isPositive();
+    }
+
+    @ParameterizedTest
+    @NullSource
+    @ValueSource(strings = {"", " ", "{}"})
+    void should_count_tokens_in_ai_message_with_empty_tool_arguments(String arguments) {
+
+        // given
+        ToolExecutionRequest toolExecutionRequest = ToolExecutionRequest.builder()
+                .id("tool-id")
+                .name("tool-name")
+                .arguments(arguments)
+                .build();
+
+        AiMessage aiMessage = AiMessage.from(toolExecutionRequest, toolExecutionRequest); // duplicate on purpose
+
+        // when
+        int tokenCount = tokenCountEstimator.estimateTokenCountInMessage(aiMessage);
+
+        // then
+        assertThat(tokenCount).isPositive();
     }
 
     @Test

